Gym Members Exercise Dataset

In [51]:
import pandas as pd
import numpy as np
df = pd.read_csv('/content/gym_members_exercise_tracking.csv')
df.head()

,Age,Gender,Weight (kg),Height (m),Max_BPM,Avg_BPM,Resting_BPM,Session_Duration (hours),Calories_Burned,Workout_Type,Fat_Percentage,Water_Intake (liters),Workout_Frequency (days/week),Experience_Level,BMI
0,56,Male,88.3,1.71,180,157,60,1.69,1313.0,Yoga,12.6,3.5,4,3,30.20
1,46,Female,74.9,1.53,179,151,66,1.30,883.0,HIIT,33.9,2.1,4,2,32.00
2,32,Female,68.1,1.66,167,122,54,1.11,677.0,Cardio,33.4,2.3,4,2,24.71
3,25,Male,53.2,1.70,190,164,56,0.59,532.0,Strength,28.8,2.1,3,1,18.41
4,38,Male,46.1,1.79,188,158,68,0.64,556.0,Strength,29.2,2.8,3,1,14.39


In [52]:
df.describe()

,Age,Weight (kg),Height (m),Max_BPM,Avg_BPM,Resting_BPM,Session_Duration (hours),Calories_Burned,Fat_Percentage,Water_Intake (liters),Workout_Frequency (days/week),Experience_Level,BMI
count,973.000000,973.000000,973.00000,973.000000,973.000000,973.000000,973.000000,973.000000,973.000000,973.000000,973.000000,973.000000,973.000000
mean,38.683453,73.854676,1.72258,179.883864,143.766701,62.223022,1.256423,905.422405,24.976773,2.626619,3.321686,1.809866,24.912127
std,12.180928,21.207500,0.12772,11.525686,14.345101,7.327060,0.343033,272.641516,6.259419,0.600172,0.913047,0.739693,6.660879
min,18.000000,40.000000,1.50000,160.000000,120.000000,50.000000,0.500000,303.000000,10.000000,1.500000,2.000000,1.000000,12.320000
25%,28.000000,58.100000,1.62000,170.000000,131.000000,56.000000,1.040000,720.000000,21.300000,2.200000,3.000000,1.000000,20.110000
50%,40.000000,70.000000,1.71000,180.000000,143.000000,62.000000,1.260000,893.000000,26.200000,2.600000,3.000000,2.000000,24.160000
75%,49.000000,86.000000,1.80000,190.000000,156.000000,68.000000,1.460000,1076.000000,29.300000,3.100000,4.000000,2.000000,28.560000
max,59.000000,129.900000,2.00000,199.000000,169.000000,74.000000,2.000000,1783.000000,35.000000,3.700000,5.000000,3.000000,49.840000


Convering the  weight and height dataset in the List

In [53]:
## Weight in the Kg::
weight = df['Weight (kg)'].tolist()
## Height in the Meter::
height = df['Height (m)'].tolist()

Couting the Number of the Sample Size

In [54]:
sample_size1 = len(df)
print(f"The Sample Size of the Height and height is {sample_size1}")

The Sample Size of the Height and height is 973


Calculating the Mean, Variance, and the Standard deviation of Weight amd Height

In [55]:
### Mean of the Weight::
mean_weight = sum(df['Weight (kg)']) / len(df['Weight (kg)'])
print(mean_weight)
### Mean of the Height ::
mean_height = sum(df['Height (m)']) / len(df['Height (m)'])
print(mean_height)

73.85467625899281
1.722579650565262


In [56]:
### Standard Deviation of the weight
sum_weight = 0
for i in weight:
  sum_weight += (i - mean_weight) ** 2
variance_weight = sum_weight / len(weight)
print(f"The Variance of the Weight is {variance_weight}")
std_weight = variance_weight ** 0.5
print(f"The Standard Deviation of the Weight is {std_weight}")
### Standard Deviation of the Height
sum_height = 0
for i in height:
  sum_height += (i - mean_height) ** 2
variance_height = sum_height / len(height)
print(f"The Variance of the height is {variance_height}")
std_height = variance_height ** 0.5
print(f"The Standard Deviation of the height is {std_height}")

The Variance of the Weight is 449.2958388725823
The Standard Deviation of the Weight is 21.19659970072045
The Variance of the height is 0.016295606451265446
The Standard Deviation of the height is 0.12765424572361644


Calculating the Standard Error of Height and Weight

In [57]:
### For the Height
standard_error_height = std_height / (sample_size1 ** 0.5)
print(f"The Standard Error of the Height is {standard_error_height}")
### For the Weight ::
standard_error_weight = std_weight / (sample_size1 ** 0.5)
print(f"The Standard Error of the Weight is {standard_error_weight}")

The Standard Error of the Height is 0.004092407234066563
The Standard Error of the Weight is 0.6795317888654713


**Conclusion **

The standard error of height (0.0040) is very small, which means the mean height is estimated very accurately. The standard error of weight (0.6795) is slightly larger, showing more variability in weight, but the mean weight estimate is still reasonably reliable due to the large sample size.

### Pearson's Correlation using Formula in Python

In [58]:

### Calculating the Sum needed for the Correlation
sum_cov = 0
sum_height_sq = 0
sum_weight_sq = 0

for i in range(sample_size1):
    diff_height = height[i] - mean_height
    diff_weight = weight[i] - mean_weight

    sum_cov += diff_height * diff_weight
    sum_height_sq += diff_height**2
    sum_weight_sq += diff_weight**2

### Calculating the Correlation Pearosn's using formula
pearson_r = sum_cov / ( (sum_height_sq**0.5) * (sum_weight_sq**0.5) )

print("Pearson Correlation Height & Weight:", pearson_r)

Pearson Correlation Height & Weight: 0.36532120261488743


### Conclusion::

The Pearson correlation between height and weight is approximately 0.37, which indicates a moderate positive relationship. This means that, in general, as height increases, weight tends to increase as well, but the relationship is not very strong.